[make:*] use php-cs-fixer to style/lint all generated php templates (#1251)
symfony/maker-bundle: README.md

[2]: https://symfony.com/doc/current/setup/flex.html
[3]: https://symfony.com/doc/current/bundles/SymfonyMakerBundle/index.html
[4]: https://symfony.com/doc/current/contributing/code/bc.html

## Linting Generated Templates

### PHP-CS-FIXER

MakerBundle uses php-cs-fixer to enforce coding standards when generating `.php`
files. If you have `friendsofphp/php-cs-fixer` added to your project, we'll

> use the `bin/php-cs-fixer` binary and `.php-cs-fixer.dist.php` configuration file

Reviewer: ```suggestion
look for and use the `bin/php-cs-fixer` binary and `.php-cs-fixer.dist.php` configuration file
```

Reviewer: Wouldn't/shouldn't it be `vendor/bin/php-cs-fixer`?

Also, since installing php-cs-fixer into your composer.json isn't a best practice, I'm not sure we should support this. We *could* look to see if there is a global `php-cs-fixer` binary available and use that.

In all cases, if we're using their binary or their config file (regardless of if it's automatic or because they set env vars), it might be a good idea to print a small note at the top of the command's output about this.

Author: When you install via composer e.g. (composer require --dev ......), php-cs-fixer installs the binary in the `path-to-project/bin` folder https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/a2bdba33ddb7df1cedce536c3e9c3db8d73ddd3e/composer.json#L65

Looking at their readme, https://github.com/PHP-CS-Fixer/PHP-CS-Fixer#installation we could check for the binary in `tools/php-cs-fixer` instead/in addition to the above.

I didn't even think about php-cs-fixer installed globally - I'm wondering if we should add this bit in after the fact due to the different ways that the binary can be installed (manually, via composer, phive, and brew). I'd have to do some playing around to see if there are any differences in the `PATH` vars for us to use. e.g. installing manually (globally) the user has to:

- Download composer -> move it to `/usr/local/bin` on linux -> ensure thats in their system `PATH`

But you can also install globally via composer by `composer global install.... `

I'm thinking we could have a more robust method of checking for the binary using the autoloader -> checking the global path(s). 

Thoughts?

> In all cases, if we're using their binary or their config file (regardless of if it's automatic or because they set env vars), it might be a good idea to print a small note at the top of the command's output about this.

Agree'd

Author: We also need to figure out how to write tests for the global installs as I don't think that is something we can do right out of the box now - but I could be mistaken.

Reviewer: I think we can simply use this class: https://github.com/symfony/process/blob/6.2/ExecutableFinder.php

If we find `php-cs-fixer`, cool. Use it. Else, don't. It's so simple I don't think we'd need any additional tests.

Reviewer: > php-cs-fixer installs the binary in the path-to-project/bin folder https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/a2bdba33ddb7df1cedce536c3e9c3db8d73ddd3e/composer.json#L65 (

I think that means "install `php-cs-fixer` into whatever Composer's `bin` directory is", which is `vendor/bin` by default.

Author: Executable finder works for me...

![image](https://user-images.githubusercontent.com/40327885/206543343-b8049632-4243-4899-a4b4-c1f0b7d16015.png)

automatically.

> Otherwise, we use the `src/Bin/php-cs-fixer-v3.13.0.phar` and

Reviewer: ```suggestion
Otherwise, we use a `php-cs-fixer` version that is packaged with this bundle.

No need to mention the version.

Also `php-cs-fixer` probably shouldn't be in `src/Bin` but just maybe `bin/` in our repo :). Or maybe `src/Resources/... something`
```

Author: moved to `Resources/bin`

`src/Resources/config/php-cs-fixer.config.php` that are included with MakerBundle.

You can explicitly set a custom path to the `php-cs-fixer` binary and/or a 

> php-cs-fixer config file by explicitly setting their respective environment

Reviewer: ```suggestion
php-cs-fixer config file by explicitly their respective environment
```
